modules/install-ibm-cloud-configuring-the-public-subnet.adoc

```diff
@@ -0,0 +1,140 @@
+
+[id="configuring-the-public-subnet_{context}"]
+= Configuring the public subnet
+
+All of the {product-title} cluster nodes must be on the public subnet. IBM Cloud&#174; does not provide a DHCP server on the subnet. Set it up separately on the provisioner node.
+
+.Procedure
+
+. Install `dnsmasq`:
++
+[source,terminal]
+----
+$ sudo dnf install dnsmasq
+----
+
+. Set an unused IP address from the public IP range:
++
+[source,terminal]
+----
+$ UNUSEDIP=<IP-address>
+----
+
+. Open the `dnsmasq` configuration file:
++
+[source,terminal]
+----
+$ sudo vi /etc/dnsmasq.conf
+----
+
+. Add the following configuration to the `dnsmasq` configuration file:
++
+[source,configuration files]
+----
+interface=baremetal
+except-interface=lo
+bind-dynamic
+log-dhcp
+
+dhcp-range=$UNUSEDIP,$UNUSEDIP,$PUBLICCIDR
+dhcp-option=baremetal,121,0.0.0.0/0,$BAREMETAL_GATEWAY,$BOOTSTRAP_PRIV_IP,$BSHOSTIP
+
+dhcp-hostsfile=/var/lib/dnsmasq/dnsmasq.hostsfile
+----
++
+[NOTE]
+====
+The Dnsmasq configuration uses the BASH variables created in steps 12-22 of the xref:preparing-the-provisioner-node-for-openshift-install-on-ibm-cloud_{context}[Preparing the provisioner node for {product-title} installation on IBM Cloud] procedure. Re-create the BASH variables for this procedure, as needed.
+====
+
+. Obtain the list of hardware for the cluster using the IBM Cloud command line interface:
++
+[source,terminal]
+----
+$ ibmcloud sl hardware list
+----
+
+. Obtain the MAC addresses and IP addresses for each node using the IBM Cloud command line interface:
++
+[source,terminal]
+----
+$ ibmcloud sl hardware detail <ID> --output JSON | \
+  jq '.networkComponents[] | \
+  "\(.primaryIpAddress) \(.macAddress)"' | grep -v null
+----
++
+Replace `<ID>` with the ID of the node.
++
+.Example output
+[source,terminal]
+----
+"10.196.130.144 00:e0:ed:6a:ca:b4"
+"141.125.65.215 00:e0:ed:6a:ca:b5"
+----
++
+Make a note of the MAC address and IP address of the public network. Make a separate note of the MAC address of the private network, which you will use later in the `install-config.yaml` file. Repeat this procedure for each node until you have all the public MAC and IP addresses for the public `baremetal` network, and the MAC addresses of the private `provisioning` network.
+
+. Add the MAC and IP address pair of the public `baremetal` network for each node into the `dnsmasq.hostsfile` file:
++
+[source,terminal]
+----
+$ sudo vim /var/lib/dnsmasq/dnsmasq.hostsfile
+----
++
+.Example input
+[source,configuration files]
+----
+00:e0:ed:6a:ca:b5,141.125.65.215,master-0
+<MAC>,<IP>,master-1
+<MAC>,<IP>,master-2
+<MAC>,<IP>,worker-0
+<MAC>,<IP>,worker-1
+...
+----
++
+Replace `<MAC>,<IP>` with the public MAC address and public IP address of the corresponding node name.
+
+. Start `dnsmasq`:
++
+[source,terminal]
+----
+$ sudo systemctl start dnsmasq
+----
+
+. Enable `dnsmasq` so that it starts when booting the node:
++
+[source,terminal]
+----
+$ sudo systemctl enable dnsmasq
+----
+
```

Comment:
I would add verification for the dnsmasq status
$ sudo systemctl status dnsmasq
● dnsmasq.service - DNS caching server.
   Loaded: loaded (/usr/lib/systemd/system/dnsmasq.service; enabled; vendor preset: disabled)
   Active: active (running) since Tue 2021-10-05 05:04:14 CDT; 49s ago
 Main PID: 3101 (dnsmasq)
    Tasks: 1 (limit: 204038)
   Memory: 732.0K
   CGroup: /system.slice/dnsmasq.service
           └─3101 /usr/sbin/dnsmasq -k

Oct 05 05:04:14 hwprov-bs.c2.cloud.goodsquishy.com dnsmasq[3101]: listening on baremetal(#8): fe80::19fe:e97e:b705:a827%baremetal
Oct 05 05:04:14 hwprov-bs.c2.cloud.goodsquishy.com dnsmasq[3101]: started, version 2.79 cachesize 150
Oct 05 05:04:14 hwprov-bs.c2.cloud.goodsquishy.com dnsmasq[3101]: compile time options: IPv6 GNU-getopt DBus no-i18n IDN2 DHCP DHCPv6 no-Lua TFTP no-conntrack ipset auth DNSSEC loop-detect inotify
Oct 05 05:04:14 hwprov-bs.c2.cloud.goodsquishy.com dnsmasq-dhcp[3101]: DHCP, IP range 141.125.65.220 -- 141.125.65.220, lease time 2m
Oct 05 05:04:14 hwprov-bs.c2.cloud.goodsquishy.com dnsmasq-dhcp[3101]: DHCP, sockets bound exclusively to interface baremetal
Oct 05 05:04:14 hwprov-bs.c2.cloud.goodsquishy.com dnsmasq[3101]: reading /etc/resolv.conf
Oct 05 05:04:14 hwprov-bs.c2.cloud.goodsquishy.com dnsmasq[3101]: using nameserver 10.0.80.11#53
Oct 05 05:04:14 hwprov-bs.c2.cloud.goodsquishy.com dnsmasq[3101]: using nameserver 10.0.80.12#53
Oct 05 05:04:14 hwprov-bs.c2.cloud.goodsquishy.com dnsmasq[3101]: read /etc/hosts - 2 addresses
Oct 05 05:04:14 hwprov-bs.c2.cloud.goodsquishy.com dnsmasq-dhcp[3101]: read /var/lib/dnsmasq/dnsmasq.hostsfile